app/(playground)/p/[agentId]/actions.ts

```diff
@@ -392,37 +397,111 @@ export async function parse(id: FileId, name: string, blobUrl: string) {
 	const jsonString = JSON.stringify(partitionResponse.elements, null, 2);
 	const blob = new Blob([jsonString], { type: "application/json" });
 
-	await put(pathJoin(vercelBlobFileFolder, id, "partition.json"), blob, {
-		access: "public",
-		contentType: blob.type,
-	});
+	await withCountMeasurement(
+		logger,
+		async () => {
+			const result = await put(
+				pathJoin(vercelBlobFileFolder, id, "partition.json"),
+				blob,
+				{ access: "public", contentType: blob.type },
+			);
+			return {
+				blob: result,
+				size: blob.size,
+			};
+		},
+		ExternalServiceName.VercelBlob,
+		startTime,
+		VercelBlobOperation.Put,
+	);
 
+	const startTimeConvertMarkdown = Date.now();
 	const markdown = elementsToMarkdown(partitionResponse.elements ?? []);
 	const markdownBlob = new Blob([markdown], { type: "text/markdown" });
-	const vercelBlob = await put(
-		pathJoin(vercelBlobFileFolder, id, "markdown.md"),
-		markdownBlob,
-		{
-			access: "public",
-			contentType: markdownBlob.type,
+	const result = await withCountMeasurement(
+		logger,
+		async () => {
+			const result = await put(
+				pathJoin(vercelBlobFileFolder, id, "markdown.md"),
+				markdownBlob,
+				{
+					access: "public",
+					contentType: markdownBlob.type,
+				},
+			);
+			return {
+				vercelBlob: result,
+				size: markdownBlob.size,
+			};
 		},
+		ExternalServiceName.VercelBlob,
+		startTimeConvertMarkdown,
+		VercelBlobOperation.Put,
 	);
```

Comment:
It is not good to have the measurement code exposed in the application logic, so I would like to implement a put function that wraps this process, but this is fine for now.